tendermint/src/lite/verifier.rs

```diff
@@ -36,80 +37,30 @@ where
     }
 }
 
-fn validate_next_vals<H, V>(header: H, next_vals: &V) -> Result<(), Error>
-where
-    H: Header,
-    V: ValidatorSet,
-{
-    // ensure the next validators in the header matches what was supplied.
-    if header.next_validators_hash() != next_vals.hash() {
-        return Err(Error::InvalidNextValidatorSet);
-    }
-
-    Ok(())
-}
-
-/// Captures the skipping condition, i.e., it defines when we can trust the header
-/// h2 based on a known trusted state.
-/// Note that the trusted header included in the trusted state and h2 have already
-/// passed basic validation by calling  `verify`.
-/// `Error::InsufficientVotingPower`is returned when there is not enough intersection
-/// between validator sets to have skipping condition true.
-pub fn check_support<TS, SH, C, L>(
-    trusted_state: &TS,
-    h2: &SH,
-    trust_threshold: &L,
-    trusting_period: &Duration,
-    now: &SystemTime,
-) -> Result<(), Error>
-where
-    TS: TrustedState<LastHeader = SH, ValidatorSet = C::ValidatorSet>,
-    SH: SignedHeader<Commit = C>,
-    L: TrustThreshold,
-    C: Commit,
-{
-    let h1 = trusted_state.last_header();
-    let h1_next_vals = trusted_state.validators();
-
-    // TODO(EB): can we move all these checks? it seems the
-    // is_within doesn't need to happen here and the sequential check
-    // for vals hash should be part of validate_vals_and_commit. then this function
-    // is basically just verify_commit_trusting. Would need to update spec as well.
-    // In sequential case, would still need to return early or not call check_support at all.
-    is_within_trust_period(h1.header(), trusting_period, now)?;
-
-    // check the sequential case
-    if h2.header().height() == h1.header().height().increment() {
-        if h2.header().validators_hash() == h1_next_vals.hash() {
-            // It's sequential, so verify_commit_trusting would be
-            // redundant with verify since the validators are the same
-            // when we're not skipping.
-            return Ok(());
-        } else {
-            return Err(Error::InvalidNextValidatorSet);
-        }
-    }
-
-    // check if enough trusted validators signed to skip to the new height.
-    verify_commit_trusting(h1_next_vals, h2.commit(), trust_threshold)
-}
-
-/// Validate the validators and commit against the header.
+/// Validate the validators, next validators, and commit against the header.
 // TODO(EB): consider making this a method on Commit so the details are hidden,
 // and so we can remove the votes_len() method (that check would be part of the
 // methods implementation). These checks aren't reflected
 // explicitly in the spec yet, only in the sentence "Additional checks should
 // be done in the implementation to ensure header is well formed".
-fn validate_vals_and_commit<H, V, C>(header: &H, commit: &C, vals: &V) -> Result<(), Error>
+pub fn validate_signed_header_and_vals<H, V, C>(
```

Comment:
Seeing a name including an `and` is typically a sign that the name just reiterates the steps inside the function body instead of conveying the higher level utility/purpose provided.

Reply:
what would you propose as a better option? the name is currently naming all the arguments (where header and commit together comprise a signed_header). How about `validate_state` or `validate_untrusted` ?

Reply:
```suggestion
pub fn validate_untrusted<H, V, C>(
```